src/ngx_pagespeed.cc

```diff
@@ -422,6 +422,20 @@ ngx_int_t ps_set_cache_control(ngx_http_request_t* r, char* cache_control) {
   return NGX_OK;
 }
 
+StringPiece ps_get_cache_control(ngx_http_request_t* r) {
+  ngx_table_elt_t* header;
+  NgxListIterator it(&(r->headers_out.headers.part));
+  while ((header = it.Next()) != NULL) {
+    if (STR_CASE_EQ_LITERAL(header->key, "Cache-Control")) {
```

Comment:
You should really use `r->headers_out.cache_control`, otherwise you're going to miss some directives in case upstream sent multiple `Cache-Control` headers, whereas NGINX is going to send all `Cache-Control` headers downstream and (if configured) it will log combined value, covering all `Cache-Control` headers.